### PR TITLE
PLT-4752 Create screens to edit OAuth 2.0 Applications

### PR DESCRIPTION
**Pull request by n1aba**
#### Summary
Create edit page for OAuth 2.0 Applications, rewrite them to be abstract, add respective jest tests.
Rewrite integration commands to use abstract component, update unit tests.
Extend webhook tests, add missing request object to installed oauth apps.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/4752

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has server changes https://github.com/mattermost/mattermost-server/pull/7413
- [x] Has redux changes https://github.com/mattermost/mattermost-redux/pull/246
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json]

### CODE REVIEW

**Comment by n1aba** [comment]
Rebased on top of https://github.com/mattermost/mattermost-webapp/commit/06e82dadffe1a8712386abb3949155eb26b218e4

**Comment by n1aba** [comment]
Rebased on top of https://github.com/mattermost/mattermost-webapp/commit/06e82dadffe1a8712386abb3949155eb26b218e4

File: yarn.lock [hunk 1]
```diff
@@ -5063,9 +5063,9 @@ math-expression-evaluator@^1.2.14:
   version "1.2.16"
   resolved "https://registry.yarnpkg.com/math-expression-evaluator/-/math-expression-evaluator-1.2.16.tgz#b357fa1ca9faefb8e48d10c14ef2bcb2d9f0a7c9"
 
-mattermost-redux@mattermost/mattermost-redux#master:
+mattermost-redux@n1aba/mattermost-redux#edit-oauth-app:
   version "0.0.1"
-  resolved "https://codeload.github.com/mattermost/mattermost-redux/tar.gz/b09a04337301c1a3cbe8a5a7e726e305a9af4f41"
+  resolved "https://codeload.github.com/n1aba/mattermost-redux/tar.gz/c9f753c8941537041adf3c8876d096db6387de1d"
```
Comment (n1aba) [inline yarn.lock]: I'll update yarn.lock and package.json once my redux PR get merged

**Comment by n1aba** [comment]
I've resolved conflicts in yarn.lock but Test Server appears to be broken, it displays 404 to me

**Comment by n1aba** [comment]
Thank you for your feedback, looking into this

**Comment by n1aba** [comment]
Nice catch @jasonblais!

1. I put the wrong redirect link to edit page and you were waiting while the whole app bundle re-dowloads on background. Now edit page load time should be similar to the others (commands & hooks).

2. You probably won't see this warning while landing on the edit page. However you might see it in other places but it's not related to my changes. This warning is feature detection and doesn't affect anything. I'm not sure if this is thrown only once on initial app load. Possible fix might be to update to higher version of pdfjs-dist, you can find more context in following issue:
https://github.com/mozilla/pdf.js/issues/6306

**Comment by jasonblais** [comment]
Thanks! Loading works great and found no issues with editing an app.

However, now I'm getting an error after creating an OAuth 2.0 application, perhaps related to the redirect changes?

 image

**Comment by n1aba** [comment]
Opps, my fault, sorry for that

**Comment by jwilander** [comment]
Nice work @n1aba ! I'll give it a code review in the next couple days

**Comment by cpanato** [comment]
I will look into this week, since this is a big PR and need some time :)
thanks for this PR @n1aba

**Comment by jwilander** [comment]
Sorry about the delay, we're going through a release and I've been bogged down. Promise to look at it tomorrow

**Comment by n1aba** [comment]
No problem, I'll need to rebase this anyway. I'll do it in the next few days

**Comment by jwilander** [comment]
@n1aba looks like it needs one more small rebase

**Pull request merged**